### ISSUE TITLE
Yeoman Tutorial - CSS not loading

### ISSUE THREAD

**Issue by bengara** [yeoman/yeoman.io#479]
I'm trying to get the Yeoman tutorial working (http://yeoman.io/codelab.html). Following the directions, everything seemed to install and setup fine. I'm successfully getting the localhost homepage, but it's unstyled. It looks like the .css is not loading (yes, the file is there: mytodo\app\styles\main.css).

yo doctor returns: Everything looks all right!

(Brand new OS X box - nothing else installed yet.)

If anyone has suggestions I would appreciate it! I'm a bit of a newbie - getting back into web development after a 10-year hiatus.

Here is the command output per the submission guidelines:

yo --version && echo $PATH $NODE_PATH && node -e 'console.log(process.platform, process.versions)' && cat Gruntfile.js

1.4.7

/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin

darwin { http_parser: '2.3',

  node: '0.12.4',

  v8: '3.28.71.19',

  uv: '1.5.0',

  zlib: '1.2.8',

  modules: '14',

  openssl: '1.0.1m' }

// Generated on 2015-06-16 using generator-angular 0.11.1

'use strict';

// # Globbing

// for performance reasons we're only matching one level down:

// 'test/spec/{,*/}*.js'

// use this if you want to recursively match all subfolders:

// 'test/spec/**/*.js'

module.exports = function (grunt) {

  // Load grunt tasks automatically

  require('load-grunt-tasks')(grunt);

  // Time how long tasks take. Can help when optimizing build times

  require('time-grunt')(grunt);

  // Configurable paths for the application

  var appConfig = {

    app: require('./bower.json').appPath || 'app',

    dist: 'dist'

  };

  // Define the configuration for all the tasks

  grunt.initConfig({

    // Project settings

    yeoman: appConfig,

    // Watches files for changes and runs tasks based on the changed files

    watch: {

      bower: {

        files: ['bower.json'],

        tasks: ['wiredep']

      },

      js: {

        files: ['<%= yeoman.app %>/scripts/{,*/}*.js'],

        tasks: ['newer:jshint:all'],

        options: {

          livereload: '<%= connect.options.livereload %>'

        }

      },

      jsTest: {

        files: ['test/spec/{,*/}*.js'],

        tasks: ['newer:jshint:test', 'karma']

      },

      styles: {

        files: ['<%= yeoman.app %>/styles/{,*/}*.css'],

        tasks: ['newer:copy:styles', 'autoprefixer']

      },

      gruntfile: {

        files: ['Gruntfile.js']

      },

      livereload: {

        options: {

          livereload: '<%= connect.options.livereload %>'

        },

        files: [

          '<%= yeoman.app %>/{,*/}*.html',

          '.tmp/styles/{,*/}*.css',

          '<%= yeoman.app %>/images/{,*/}*.{png,jpg,jpeg,gif,webp,svg}'

        ]

      }

    },

    // The actual grunt server settings

    connect: {

      options: {

        port: 9000,

        // Change this to '0.0.0.0' to access the server from outside.

        hostname: 'localhost',

        livereload: 35729

      },

      livereload: {

        options: {

          open: true,

          middleware: function (connect) {

            return [

              connect.static('.tmp'),

              connect().use(

                '/bower_components',

                connect.static('./bower_components')

              ),

              connect().use(

                '/app/styles',

                connect.static('./app/styles')

              ),

              connect.static(appConfig.app)

            ];

          }

        }

      },

      test: {

        options: {

          port: 9001,

          middleware: function (connect) {

            return [

              connect.static('.tmp'),

              connect.static('test'),

              connect().use(

                '/bower_components',

                connect.static('./bower_components')

              ),

              connect.static(appConfig.app)

            ];

          }

        }

      },

      dist: {

        options: {

          open: true,

          base: '<%= yeoman.dist %>'

        }

      }

    },

    // Make sure code styles are up to par and there are no obvious mistakes

    jshint: {

      options: {

        jshintrc: '.jshintrc',

        reporter: require('jshint-stylish')

      },

      all: {

        src: [

          'Gruntfile.js',

          '<%= yeoman.app %>/scripts/{,*/}*.js'

        ]

      },

      test: {

        options: {

          jshintrc: 'test/.jshintrc'

        },

        src: ['test/spec/{,*/}*.js']

      }

    },

    // Empties folders to start fresh

    clean: {

      dist: {

        files: [{

          dot: true,

          src: [

            '.tmp',

            '<%= yeoman.dist %>/{,*/}*',

            '!<%= yeoman.dist %>/.git{,*/}*'

          ]

        }]

      },

      server: '.tmp'

    },

    // Add vendor prefixed styles

    autoprefixer: {

      options: {

        browsers: ['last 1 version']

      },

      server: {

        options: {

          map: true,

        },

        files: [{

          expand: true,

          cwd: '.tmp/styles/',

          src: '{,*/}*.css',

          dest: '.tmp/styles/'

        }]

      },

      dist: {

        files: [{

          expand: true,

          cwd: '.tmp/styles/',

          src: '{,*/}*.css',

          dest: '.tmp/styles/'

        }]

      }

    },

    // Automatically inject Bower components into the app

    wiredep: {

      app: {

        src: ['<%= yeoman.app %>/index.html'],

        ignorePath:  /\.\.\//

      },

      test: {

        devDependencies: true,

        src: '<%= karma.unit.configFile %>',

        ignorePath:  /\.\.\//,

        fileTypes:{

          js: {

            block: /(([\s\t]*)\/{2}\s*?bower:\s*?(\S*))(\n|\r|.)*?(\/{2}\s*endbower)/gi,

              detect: {

                js: /'(.*\.js)'/gi

              },

              replace: {

                js: '\'{{filePath}}\','

              }

            }

          }

      }

    },

    // Renames files for browser caching purposes

    filerev: {

      dist: {

        src: [

          '<%= yeoman.dist %>/scripts/{,*/}*.js',

          '<%= yeoman.dist %>/styles/{,*/}*.css',

          '<%= yeoman.dist %>/images/{,*/}*.{png,jpg,jpeg,gif,webp,svg}',

          '<%= yeoman.dist %>/styles/fonts/*'

        ]

      }

    },

    // Reads HTML for usemin blocks to enable smart builds that automatically

    // concat, minify and revision files. Creates configurations in memory so

    // additional tasks can operate on them

    useminPrepare: {

      html: '<%= yeoman.app %>/index.html',

      options: {

        dest: '<%= yeoman.dist %>',

        flow: {

          html: {

            steps: {

              js: ['concat', 'uglifyjs'],

              css: ['cssmin']

            },

            post: {}

          }

        }

      }

    },

    // Performs rewrites based on filerev and the useminPrepare configuration

    usemin: {

      html: ['<%= yeoman.dist %>/{,*/}*.html'],

      css: ['<%= yeoman.dist %>/styles/{,*/}*.css'],

      options: {

        assetsDirs: [

          '<%= yeoman.dist %>',

          '<%= yeoman.dist %>/images',

          '<%= yeoman.dist %>/styles'

        ]

      }

    },

    // The following *-min tasks will produce minified files in the dist folder

    // By default, your `index.html`'s  will take care of

    // minification. These next options are pre-configured if you do not wish

    // to use the Usemin blocks.

    // cssmin: {

    //   dist: {

    //     files: {

    //       '<%= yeoman.dist %>/styles/main.css': [

    //         '.tmp/styles/{,*/}*.css'

    //       ]

    //     }

    //   }

    // },

    // uglify: {

    //   dist: {

    //     files: {

    //       '<%= yeoman.dist %>/scripts/scripts.js': [

    //         '<%= yeoman.dist %>/scripts/scripts.js'

    //       ]

    //     }

    //   }

    // },

    // concat: {

    //   dist: {}

    // },

    imagemin: {

      dist: {

        files: [{

          expand: true,

          cwd: '<%= yeoman.app %>/images',

          src: '{,*/}*.{png,jpg,jpeg,gif}',

          dest: '<%= yeoman.dist %>/images'

        }]

      }

    },

    svgmin: {

      dist: {

        files: [{

          expand: true,

          cwd: '<%= yeoman.app %>/images',

          src: '{,*/}*.svg',

          dest: '<%= yeoman.dist %>/images'

        }]

      }

    },

    htmlmin: {

      dist: {

        options: {

          collapseWhitespace: true,

          conservativeCollapse: true,

          collapseBooleanAttributes: true,

          removeCommentsFromCDATA: true,

          removeOptionalTags: true

        },

        files: [{

          expand: true,

          cwd: '<%= yeoman.dist %>',

          src: ['*.html', 'views/{,*/}*.html'],

          dest: '<%= yeoman.dist %>'

        }]

      }

    },

    // ng-annotate tries to make the code safe for minification automatically

    // by using the Angular long form for dependency injection.

    ngAnnotate: {

      dist: {

        files: [{

          expand: true,

          cwd: '.tmp/concat/scripts',

          src: '*.js',

          dest: '.tmp/concat/scripts'

        }]

      }

    },

    // Replace Google CDN references

    cdnify: {

      dist: {

        html: ['<%= yeoman.dist %>/*.html']

      }

    },

    // Copies remaining files to places other tasks can use

    copy: {

      dist: {

        files: [{

          expand: true,

          dot: true,

          cwd: '<%= yeoman.app %>',

          dest: '<%= yeoman.dist %>',

          src: [

            '*.{ico,png,txt}',

            '.htaccess',

            '*.html',

            'views/{,*/}*.html',

            'images/{,*/}*.{webp}',

            'styles/fonts/{,*/}*.*'

          ]

        }, {

          expand: true,

          cwd: '.tmp/images',

          dest: '<%= yeoman.dist %>/images',

          src: ['generated/*']

        }, {

          expand: true,

          cwd: 'bower_components/bootstrap/dist',

          src: 'fonts/*',

          dest: '<%= yeoman.dist %>'

        }]

      },

      styles: {

        expand: true,

        cwd: '<%= yeoman.app %>/styles',

        dest: '.tmp/styles/',

        src: '{,*/}*.css'

      }

    },

    // Run some tasks in parallel to speed up the build process

    concurrent: {

      server: [

        'copy:styles'

      ],

      test: [

        'copy:styles'

      ],

      dist: [

        'copy:styles',

        'imagemin',

        'svgmin'

      ]

    },

    // Test settings

    karma: {

      unit: {

        configFile: 'test/karma.conf.js',

        singleRun: true

      }

    }

  });

  grunt.registerTask('serve', 'Compile then start a connect web server', function (target) {

    if (target === 'dist') {

      return grunt.task.run(['build', 'connect:dist:keepalive']);

    }

    grunt.task.run([

      'clean:server',

      'wiredep',

      'concurrent:server',

      'autoprefixer:server',

      'connect:livereload',

      'watch'

    ]);

  });

  grunt.registerTask('server', 'DEPRECATED TASK. Use the "serve" task instead', function (target) {

    grunt.log.warn('The `server` task has been deprecated. Use `grunt serve` to start a server.');

    grunt.task.run(['serve:' + target]);

  });

  grunt.registerTask('test', [

    'clean:server',

    'wiredep',

    'concurrent:test',

    'autoprefixer',

    'connect:test',

    'karma'

  ]);

  grunt.registerTask('build', [

    'clean:dist',

    'wiredep',

    'useminPrepare',

    'concurrent:dist',

    'autoprefixer',

    'concat',

    'ngAnnotate',

    'copy:dist',

    'cdnify',

    'cssmin',

    'uglify',

    'filerev',

    'usemin',

    'htmlmin'

  ]);

  grunt.registerTask('default', [

    'newer:jshint',

    'test',

    'build'

  ]);

};

**Comment by criles25** [comment]
I'm having the same problem. Try running it and saying Yes to using Sass and Yes to using Bootstrap.

**Comment by bengara** [comment]
Actually, it IS loading the main.css file. The style variation between styled and unstyled is so subtle I didn't notice it. In any case, it's not looking at all like the screenshots from the tutorial.

I'm guessing it's supposed to be relying on another stylesheet somewhere like @criles25 mentioned - Bootstrap; and I'm guessing somehow my install didn't grab it. I will try again.

**Comment by juame** [comment]
+1! Just started with yeoman, angular and bootstrap...
My workaround: Modify bower.json to use bootstrap release 3.2.x instead of the latest one: 3.3.5:
```
$ diff bower.json bower.json.bak
6c6
<     "bootstrap": "3.2.0",
---
>     "bootstrap": "^3.2.0",
```
Then ```bower install```

**Comment by arthurvr** [comment]
Chances are hight this is a duplicate of https://github.com/yeoman/generator-angular/issues/1116. Can somebody confirm that?

**Comment by addyosmani** [comment]
Afaik, this is a dupe of that issue but it would really be useful to get a few more confirmations.

**Comment by 47ronin** [comment]
You can do a simple workaround for this issue by getting back into your terminal, in the project directory: `bower install --save bootstrap#3.3.4` and then `grunt wiredep`

**Comment by jasonwhite1976** [comment]
Thanks 47ronin - was having the same problem. Working now in OSX - I'm sure it will work for win7 too.

**Comment by arthurvr** [comment]
Closing as a [duplicate](https://github.com/yeoman/generator-angular/issues/1116#issuecomment-113263353).

**Comment by lovezapp** [comment]
thanks, @47ronin ! that solution worked for me, too.

**Comment by robachma** [comment]
thanks, @47ronin - clutch work

**Comment by vb2807** [comment]
Thank you @47ronin 

Your solution works
